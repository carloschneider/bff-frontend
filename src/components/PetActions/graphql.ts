import { gql } from '@apollo/client'

export type CheckType = {
  id: string
}

export type CheckInDataType = {
  checkIn: CheckType[]
}

export type CheckVariablesType = {
  petId: string
}

export const MUTATION_CHECKIN = gql`
  mutation CheckIn($petId: String!) {
    checkIn(petId: $petId) {
      id
    }
  }
`

export type CheckOutDataType = {
  checkIn: CheckType[]
}

export const MUTATION_CHECKOUT = gql`
  mutation CheckOut($petId: String!) {
    checkOut(petId: $petId) {
      id
    }
  }
`
