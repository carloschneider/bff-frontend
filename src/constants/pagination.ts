import { OrderEnum } from 'helpers/pagination/order'

export const PAGINATION_DEFAULT_PAGE = 1
export const PAGINATION_DEFAULT_ORDER = OrderEnum.DESC
export const PAGINATION_DEFAULT_FIELD = 'createdAt'
export const PAGINATION_DEFAULT_LIMIT = 10
export const PAGINATION_DEFAULT_OFFSET = 0
