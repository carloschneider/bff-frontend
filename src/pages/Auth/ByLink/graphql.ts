import { gql } from '@apollo/client'

export type AuthByKeyType = {
  token: string
  role: string
}

export type AuthByKeyDataType = {
  authByKey: AuthByKeyType
}

export type AuthByKeyInputType = {
  key: string
}

export type AuthByKeyVariablesInputType = {
  input: AuthByKeyInputType
}

export const MUTATION_AUTH_BY_KEY = gql`
  mutation AuthByKeyInput($input: AuthByKeyInput!) {
    authByKey(input: $input) {
      token
      role
    }
  }
`
