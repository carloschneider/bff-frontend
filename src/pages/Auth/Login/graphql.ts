import { gql } from '@apollo/client'

export type AuthTutorType = {
  id: string
}

export type AuthTutorDataType = {
  authTutor: AuthTutorType
}

export type AuthTutorVariablesType = {
  email: string
  company?: string
}

export type AuthTutorVariablesInputType = {
  input: AuthTutorVariablesType
}

export const MUTATION_AUTH_TUTOR = gql`
  mutation AuthTutor($input: EmailAuthInput!) {
    authTutor(input: $input) {
      id
    }
  }
`
