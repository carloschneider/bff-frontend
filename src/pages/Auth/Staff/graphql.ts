import { gql } from '@apollo/client'

export type AuthStaffType = {
  id: string
}

export type AuthStaffDataType = {
  authStaff: AuthStaffType
}

export type AuthStaffVariablesType = {
  email: string
  company?: string
}

export type AuthStaffVariablesInputType = {
  input: AuthStaffVariablesType
}

export const MUTATION_AUTH_STAFF = gql`
  mutation AuthStaff($input: EmailAuthInput!) {
    authStaff(input: $input) {
      id
    }
  }
`
