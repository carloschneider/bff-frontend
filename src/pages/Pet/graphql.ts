import { gql } from '@apollo/client'

// Breed
export type BreedType = {
  name: string
}

// Tutor
export type TutorType = {
  id: string
  firstName: string
  lastName: string
  phoneNumber: string
}

export type PetType = {
  id: string
  name: string
  birthDate: Date
  breed: BreedType
  count?: number | null
  tutors: TutorType[]
  createdAt?: Date
}

export type PetDataType = {
  getPetById: PetType
}

export type PetVariablesType = {
  petId?: string
}

export const GET_PET_BY_ID = gql`
  query GetPetById($petId: String!) {
    getPetById(petId: $petId) {
      id
      name
      birthDate
      breed {
        name
      }
      tutors {
        id
        firstName
        lastName
        phoneNumber
      }
    }
  }
`
