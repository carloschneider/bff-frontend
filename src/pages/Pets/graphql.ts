import { gql } from '@apollo/client'

import { OrderEnum } from 'helpers/pagination/order'
import { PetType } from 'pages/Pet/graphql'

export type PetsDataType = {
  getAllPets: Omit<PetType, 'tutors'>[]
}

export type PaginationType = {
  order?: OrderEnum
  field?: keyof PetType
  limit: number
  offset: number
}

type PetsWhereType = {
  name?: string
}

export type PetsVariablesType = {
  where?: PetsWhereType | null
} & PaginationType

export const QUERY_GET_ALL_PETS = gql`
  query GetAllPets(
    $limit: Int
    $offset: Int
    $field: String
    $order: String
    $where: PetWhereInput
  ) {
    getAllPets(
      limit: $limit
      offset: $offset
      field: $field
      order: $order
      where: $where
    ) {
      id
      name
      birthDate
      createdAt
      breed {
        name
      }
      count
    }
  }
`
