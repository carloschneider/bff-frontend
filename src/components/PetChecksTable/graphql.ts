import { gql } from '@apollo/client'

import { PaginationType } from 'pages/Pets'

export type StaffType = {
  id: string
  firstName: string
  lastName: string
}

export type UserType = {
  id: string
  companyName: string
}

export type ResponsibleType = {
  id: string
  type: 'arrive' | 'leave'
  staff: StaffType | null
  user: UserType | null
}

export type CheckType = {
  id: string
  arrive: string
  leave: string | null
  count?: number | null
  responsibles: ResponsibleType[]
}

export type ChecksDataType = {
  getAllChecksByPetId: CheckType[] | undefined
}

type ChecksWhereType = {
  id?: string
}

export type ChecksVariablesType = {
  where?: ChecksWhereType | null
} & PaginationType

export const GET_CHECKS_BY_PET_ID = gql`
  query GetAllChecksByPetId(
    $limit: Int
    $offset: Int
    $field: String
    $order: String
    $where: CheckWhereGetAllByPetIdInput!
  ) {
    getAllChecksByPetId(
      limit: $limit
      offset: $offset
      field: $field
      order: $order
      where: $where
    ) {
      id
      arrive
      leave
      count
      responsibles {
        id
        type
        staff {
          id
          firstName
          lastName
        }
        user {
          id
          companyName
        }
      }
    }
  }
`
