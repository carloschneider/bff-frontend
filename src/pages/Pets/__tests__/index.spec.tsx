import { MockedResponse } from '@apollo/react-testing'
import { act, fireEvent, screen, within } from '@testing-library/react'
import { App } from 'antd'
import { GraphQLError } from 'graphql'

import { OrderEnum } from 'helpers/pagination/order'
import { renderWithApollo } from 'test-utils/render/renderWithApollo'

import PagePets from '..'
import {
  getAllPetsFixture,
  getAllPetsOrderFixture,
  getAllPetsPaginationFixture,
  getAllPetsPaginationPageTwoFixture,
  getAllPetsSearchResultFixture
} from '../__fixtures__'
import { PetsDataType, QUERY_GET_ALL_PETS } from '../graphql'

jest.mock('react-router-dom', () => ({
  Link: ({ children }: React.PropsWithChildren<Record<never, never>>) =>
    children
}))

jest.mock('constants/pagination', () => ({
  ...jest.requireActual('constants/pagination'),
  PAGINATION_DEFAULT_LIMIT: 2
}))

describe('<PagePets />', () => {
  const initialRequest = {
    request: {
      query: QUERY_GET_ALL_PETS,
      variables: {
        order: OrderEnum.DESC,
        field: 'createdAt',
        limit: 2,
        offset: 0,
        where: null
      }
    },
    result: {
      data: {
        getAllPets: getAllPetsFixture
      }
    }
  }

  const apolloResponseMock: MockedResponse<PetsDataType>[] = [initialRequest]

  it('should render component', async () => {
    const { container } = renderWithApollo(
      <App>
        <PagePets />
      </App>,
      apolloResponseMock
    )

    const petName = await screen.findByText('Fake Pet Name 1')
    expect(petName).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should search a pet', async () => {
    const mockedSearch = 'search pet'
    const apolloResponseMockSearch: MockedResponse<PetsDataType>[] = [
      initialRequest,
      {
        request: {
          query: QUERY_GET_ALL_PETS,
          variables: {
            order: OrderEnum.DESC,
            field: 'createdAt',
            limit: 2,
            offset: 0,
            where: {
              name: mockedSearch
            }
          }
        },
        result: {
          data: {
            getAllPets: getAllPetsSearchResultFixture
          }
        }
      },
      initialRequest
    ]

    renderWithApollo(
      <App>
        <PagePets />
      </App>,
      apolloResponseMockSearch
    )

    let petName = await screen.findByText('Fake Pet Name 1')
    expect(petName).toBeInTheDocument()

    let searchInput = screen.getByPlaceholderText('Search by pet name')

    await fireEvent.change(searchInput, {
      target: {
        value: mockedSearch
      }
    })

    const resultPetName = await screen.findByText(mockedSearch)
    expect(resultPetName).toBeInTheDocument()

    searchInput = screen.getByPlaceholderText('Search by pet name')

    await fireEvent.change(searchInput, {
      target: {
        value: ''
      }
    })

    petName = await screen.findByText('Fake Pet Name 1')
    expect(petName).toBeInTheDocument()
  })

  it('should sort data when click in a column', async () => {
    const apolloResponseMockSort: MockedResponse<PetsDataType>[] = [
      initialRequest,
      {
        request: {
          query: QUERY_GET_ALL_PETS,
          variables: {
            order: OrderEnum.ASC,
            field: 'createdAt',
            limit: 2,
            offset: 0,
            where: null
          }
        },
        result: {
          data: {
            getAllPets: getAllPetsOrderFixture
          }
        }
      }
    ]

    const { baseElement } = renderWithApollo(
      <App>
        <PagePets />
      </App>,
      apolloResponseMockSort
    )

    let petName = await screen.findByText('Fake Pet Name 1')
    expect(petName).toBeInTheDocument()

    const rowsBeforeSort =
      baseElement.querySelectorAll<HTMLElement>('.ant-table-row')

    expect(
      within(rowsBeforeSort[0]).getByText('Fake Pet Name 1')
    ).toBeInTheDocument()
    expect(
      within(rowsBeforeSort[1]).getByText('Fake Pet Name 2')
    ).toBeInTheDocument()

    const sorter = baseElement.querySelector('.ant-table-column-sort')

    if (sorter) {
      await fireEvent.click(sorter)
    }

    petName = await screen.findByText('Fake Pet Name 1')
    expect(petName).toBeInTheDocument()

    const rowsAfterSort =
      baseElement.querySelectorAll<HTMLElement>('.ant-table-row')

    expect(
      within(rowsAfterSort[0]).getByText('Fake Pet Name 2')
    ).toBeInTheDocument()
    expect(
      within(rowsAfterSort[1]).getByText('Fake Pet Name 1')
    ).toBeInTheDocument()
  })

  it('should change pagination page', async () => {
    const apolloResponseMockPagination: MockedResponse<PetsDataType>[] = [
      {
        request: {
          query: QUERY_GET_ALL_PETS,
          variables: {
            order: OrderEnum.DESC,
            field: 'createdAt',
            limit: 2,
            offset: 0,
            where: null
          }
        },
        result: {
          data: {
            getAllPets: getAllPetsPaginationFixture
          }
        }
      },
      {
        request: {
          query: QUERY_GET_ALL_PETS,
          variables: {
            order: OrderEnum.DESC,
            field: 'createdAt',
            limit: 2,
            offset: 2,
            where: null
          }
        },
        result: {
          data: {
            getAllPets: getAllPetsPaginationPageTwoFixture
          }
        }
      }
    ]

    renderWithApollo(
      <App>
        <PagePets />
      </App>,
      apolloResponseMockPagination
    )

    const petName = await screen.findByText('Fake Pet Name 1')
    expect(petName).toBeInTheDocument()

    const nextPageButton = screen.getByRole('button', {
      name: /right/
    })

    await fireEvent.click(nextPageButton)

    const petNameNextPage = await screen.findByText('Fake Pet Name 3')
    expect(petNameNextPage).toBeInTheDocument()
  })

  it('should show a error message', async () => {
    const apolloResponseMockError: MockedResponse<PetsDataType>[] = [
      {
        request: {
          query: QUERY_GET_ALL_PETS,
          variables: {
            order: OrderEnum.DESC,
            field: 'createdAt',
            limit: 2,
            offset: 0,
            where: null
          }
        },
        result: {
          errors: [new GraphQLError('Mocked error message')]
        }
      }
    ]

    renderWithApollo(
      <App>
        <PagePets />
      </App>,
      apolloResponseMockError
    )

    await act(() => new Promise((done) => setTimeout(done, 100)))

    const errorMessage = screen.getByText('Mocked error message')

    expect(errorMessage).toBeInTheDocument()
  })
})
