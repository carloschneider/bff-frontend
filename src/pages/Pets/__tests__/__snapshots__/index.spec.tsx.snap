// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PagePets /> should render component 1`] = `
<div
  class="css-dev-only-do-not-override-1n7nwfa ant-app"
>
  <h2
    class="ant-typography css-dev-only-do-not-override-1n7nwfa"
  >
    Pets
  </h2>
  <div
    class="ant-space css-dev-only-do-not-override-1n7nwfa ant-space-vertical"
    style="width: 100%;"
  >
    <div
      class="ant-space-item"
      style="margin-bottom: 24px;"
    >
      <span
        class="ant-input-affix-wrapper ant-input-affix-wrapper-lg css-dev-only-do-not-override-1n7nwfa"
      >
        <span
          class="ant-input-prefix"
        >
          <span
            aria-label="search"
            class="anticon anticon-search"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="search"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
              />
            </svg>
          </span>
        </span>
        <input
          class="ant-input ant-input-lg css-dev-only-do-not-override-1n7nwfa"
          placeholder="Search by pet name"
          type="search"
          value=""
        />
      </span>
    </div>
    <div
      class="ant-space-item"
    >
      <div
        class="ant-table-wrapper css-dev-only-do-not-override-1n7nwfa"
      >
        <div
          class="ant-spin-nested-loading css-dev-only-do-not-override-1n7nwfa"
        >
          <div>
            <div
              aria-busy="true"
              aria-live="polite"
              class="ant-spin ant-spin-spinning css-dev-only-do-not-override-1n7nwfa"
            >
              <span
                class="ant-spin-dot ant-spin-dot-spin"
              >
                <i
                  class="ant-spin-dot-item"
                />
                <i
                  class="ant-spin-dot-item"
                />
                <i
                  class="ant-spin-dot-item"
                />
                <i
                  class="ant-spin-dot-item"
                />
              </span>
            </div>
          </div>
          <div
            class="ant-spin-container ant-spin-blur"
          >
            <div
              class="ant-table ant-table-scroll-horizontal"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                  style="overflow-x: auto; overflow-y: hidden;"
                >
                  <table
                    style="width: 800px; min-width: 100%; table-layout: auto;"
                  >
                    <colgroup>
                      <col />
                      <col
                        style="width: 200px;"
                      />
                      <col
                        style="width: 200px;"
                      />
                      <col
                        style="width: 200px;"
                      />
                    </colgroup>
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          aria-label="Name"
                          class="ant-table-cell ant-table-column-has-sorters"
                          scope="col"
                          tabindex="0"
                        >
                          <div
                            class="ant-table-column-sorters"
                          >
                            <span
                              class="ant-table-column-title"
                            >
                              Name
                            </span>
                            <span
                              class="ant-table-column-sorter ant-table-column-sorter-full"
                            >
                              <span
                                class="ant-table-column-sorter-inner"
                              >
                                <span
                                  aria-label="caret-up"
                                  class="anticon anticon-caret-up ant-table-column-sorter-up"
                                  role="presentation"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="caret-up"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="0 0 1024 1024"
                                    width="1em"
                                  >
                                    <path
                                      d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                    />
                                  </svg>
                                </span>
                                <span
                                  aria-label="caret-down"
                                  class="anticon anticon-caret-down ant-table-column-sorter-down"
                                  role="presentation"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="caret-down"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="0 0 1024 1024"
                                    width="1em"
                                  >
                                    <path
                                      d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </span>
                          </div>
                        </th>
                        <th
                          class="ant-table-cell"
                          scope="col"
                        >
                          Breed
                        </th>
                        <th
                          class="ant-table-cell"
                          scope="col"
                        >
                          Birth date
                        </th>
                        <th
                          aria-sort="descending"
                          class="ant-table-cell ant-table-column-sort ant-table-column-has-sorters"
                          scope="col"
                          tabindex="0"
                        >
                          <div
                            class="ant-table-column-sorters"
                          >
                            <span
                              class="ant-table-column-title"
                            >
                              Created at
                            </span>
                            <span
                              class="ant-table-column-sorter ant-table-column-sorter-full"
                            >
                              <span
                                class="ant-table-column-sorter-inner"
                              >
                                <span
                                  aria-label="caret-up"
                                  class="anticon anticon-caret-up ant-table-column-sorter-up"
                                  role="presentation"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="caret-up"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="0 0 1024 1024"
                                    width="1em"
                                  >
                                    <path
                                      d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                    />
                                  </svg>
                                </span>
                                <span
                                  aria-label="caret-down"
                                  class="anticon anticon-caret-down ant-table-column-sorter-down active"
                                  role="presentation"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="caret-down"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="0 0 1024 1024"
                                    width="1em"
                                  >
                                    <path
                                      d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </span>
                          </div>
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        aria-hidden="true"
                        class="ant-table-measure-row"
                        style="height: 0px; font-size: 0px;"
                      >
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                        <td
                          style="padding: 0px; border: 0px; height: 0px;"
                        >
                          <div
                            style="height: 0px; overflow: hidden;"
                          >
                             
                          </div>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="c504316d-5d49-4a8e-8d3f-0e47e94a8694"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          Fake Pet Name 1
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          Fake Breed 1
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          10/11/2018
                        </td>
                        <td
                          class="ant-table-cell ant-table-column-sort"
                        >
                          30/12/2022 14:45:40
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="54737ecb-ed7e-4f5a-b645-25e7033f5fc2"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          Fake Pet Name 2
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          Fake Breed 2
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          10/11/2018
                        </td>
                        <td
                          class="ant-table-cell ant-table-column-sort"
                        >
                          30/12/2022 14:45:38
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
